fe/projects/Seller.Portal/src/areas/Products/components/ProductCardForm/ProductCardForm.js

```diff
@@ -0,0 +1,447 @@
+            method: "GET",
+            headers: { 
+                "Content-Type": "application/json"
+            }
+        };
+
+        fetch(`${props.definitionUrl}/${id}`, requestOptions)
+            .then((response) => {
+                dispatch({ type: "SET_IS_LOADING", payload: false });
+
+                AuthenticationHelper.HandleResponse(response);
+
+                return response.json().then(jsonResponse => {
+                    if (response.ok) {
+                        const newDefinitions = definitions;
+
+                        newDefinitions[id] = jsonResponse.data;
+
+                        setDefinitions(newDefinitions);
+                    }
+                    else {
+                        toast.error(props.generalErrorMessage);
+                    }
+                })
+            });
+    }
+
+    const handleSaveCard = (schema) => {
+        const newElements = generateElementsFromSchema(schema);
+
+        if (productAttribute != null) {
+
+            let newElement = {
+                name: productAttribute.name,
+                required: false,
+                dataOptions: {
+                    title: productAttribute.title,
+                    type: productAttribute.type,
+                    default: ""
+                }
+            }
+
+            if (productAttribute.definitionId !== undefined) {
+                newElement = {
+                    ...newElement,
+                    definitionId: productAttribute.definitionId
+                }
+
+                handleDefinitionSchema(productAttribute.definitionId);
+            }
+
+            const existingProductAttributeIndex = newElements.findIndex((element) => element.name === productAttribute.name);
+
+            if (existingProductAttributeIndex != -1) {
+                newElements[existingProductAttributeIndex] = newElement;
+            }
+            else {
+                newElements.splice(0, 0, newElement)
+            }
+
+            updateSchema(newElements, schema)
+    
+            setIsModalOpen(false);
+            setProductAttribute(null);
+        } 
+    }
+
+    const handleDeleteDialogClose = () => {
+        setOpenDeleteDialog(false);
+        setEntityToDelete(null);
+    };
+
+    const handleDeleteClick = (item) => {
+        setEntityToDelete(item);
+        setOpenDeleteDialog(true);
+    };
+
+    const handleDeleteEntity = () => {
+        const newElements = generateElementsFromSchema(schema);
+
+        const index = newElements.findIndex((element) => element.name === entityToDelete.name);
+
+        const element = newElements[index];
+
+        if (element.definitionId !== undefined) {
+            const definitionItems = newElements.filter((item) => item.definitionId === element.definitionId);
+
+            if (!(definitionItems.length > 1)) {
+                const newDefinitions = definitions;
+
+                delete newDefinitions[element.definitionId];
```

Comment:
**suggestion:** delete is dangerous, wouldn't we be able to use immutable technique to do it